### ISSUE TITLE
Post-upgrade all feeds gone

### ISSUE THREAD

**Issue by rob-stewart** [brief-rss/brief#431]
After allowing FF to upgrade yesterday to 65.0, all of my Brief feeds are gone. Can you help me recover my feeds?

FF had been current before, so I had been running whatever the latest 64.x was. Brief is v2.5.9. I don't know if that upgraded at the same time.

I still have bookmarks referring to the feeds I was subscribed to, though I don't know whether Brief still refers to those. The point is, I could recreate most (all?) of my subscriptions, but my history would be lost.

**Comment by VillageCatDad** [comment]
Brief is sensitive to FF profile issues.  I export my Brief feeds at least once a week, and when I get  a problem, I first try to delete Brief and reinstall, then re-import my feeds.  Have to clear out a lot of old articles from the Brief Trash Folder, but that's ok.  Sometimes, it's so bad that I have to copy the relevant files from my FF profile, and do a full re-install (new FF profile, then all my add-on extensions).  I'm guessing that your FF profile has been corrupted, and you may need to start over.  Best of luck!

**Comment by rob-stewart** [comment]
Thanks. I never even noticed that I could export my Brief feeds, so I'll start doing that. In this case, after a reboot of my computer, not just restarts of FF, I found that my Brief feeds were partially back. I think all of the feeds were present, but the content was not entirely restored to my previous state. That's much better than before, but it's troubling that so much was actually available, but Brief couldn't show it to me until I had restarted FF enough times or rebooted my computer.

It would be nice if Brief was better at managing its state. I realize that if the FF profile is corrupted, things are largely (entirely?) out of Brief's hands. However, some sort of auto-export-and-recover feature would be nice.

**Comment by VillageCatDad** [comment]
Yes, I agree, export-and-recover would be nice.  Also, because I have to re-import my Brief feeds fairly regularly, a way to globally set all feeds to "Don't mark changed items as unread" at once would save a lot time (now, I have right click on the first feed to get to the feed settings dialog box, and manually click a box, one feed at a time.

**Comment by Sxderp** [comment]
Another issue is WebExtensions. Handling data is no longer within the hands of extensions, we have  to trust in Firefox to do it properly. Furthermore, if Firefox even 'thinks' that an extension is gone it will delete the data with very little possibility of recovery as there are no files in your profile directory.

**Comment by MorganChampion** [comment]
Another issue is, unlike Feedbro, you can't put the various feeds into folders.

"You can't dislike a 14-second video and decide as a result you're entitled to kill the American ambassador.
It's not acceptable. It just has to stop."
Salman Rushdie, The Listener, October 13-19 2012, pg 18.
 

    On Tuesday, 5 February 2019, 8:52:17 am NZDT, Sxderp <notifications@github.com> wrote:  
 
 
Another issue is WebExtensions. Handling data is no longer within the hands of extensions, we have to trust in Firefox to do it properly. Furthermore, if Firefox even 'thinks' that an extension is gone it will delete the data with very little possibility of recovery as there are no files in your profile directory.

**Comment by VillageCatDad** [comment]
MorganChampion -- Brief has folders. It's not obvious, but straightforward.  Click on "Organize".  At the bottom of your feed list, there should be an ellipse ("...").  Click on that, and type in your folder name.  Then start dragging feeds into that folder.  Rinse and repeat. :-)  FYI - I have 225+ feeds in 14 folders.

**Comment by VillageCatDad** [comment]
Alternately, you can use a text editor...Export your feeds to an OPML file.  Open that file with your favorite text editor.  The XML tag you will need is 

   <outline text="Folder name here">
         <outline text="feed names here"..../>
   </outline>

Personally, I found the initial organizing of my 200+ feeds faster this way.  Now, since I only create new folders once in a while, I use the "Organize" feature.  Hope that helps!

**Comment by VillageCatDad** [comment]
Oh, forgot to finish!  You should Import the Feeds OPML back into Brief, and make sure it works.  Once that is done, delete all your feeds in Brief, and import the OPML file back in one more time.

**Comment by VillageCatDad** [comment]
Ok, struggling with XML embedded with comments.  Trying again.

If you export your feeds to an OPML file, you can edit it.  (Replace the parenthesis below with angle brackets)

   (outline text="folder name here")
        (outline text="feed name here" plus other attributes /)
   (/outline)

I found this easier/faster to organize my 200+ feeds initially, rather than using the one at a time movement in Organize mode.  Now that I'm done and only occasionally create new folders or move feeds, I use Organize.

Once you're done editing, save the OPML file, and Import the feeds back into Brief.  Once you're happy with the results, delete all the feeds, and import back one last time for a "clean" organization.  You probably will have to clean up your feed history (Trash folder) a time or two before you get caught back up with current posts.

Hope that helps!

**Comment by rob-stewart** [comment]
@VillageCatDad, I wanted to export my feeds as you had suggested, but when I click on Export... in the Organize menu (which never opened before), nothing happens. Import... opens a dialog though.

Will putting feeds into folders provide any more safety for the data?
--
Rob

(Sent from my portable computation device.)

**Comment by VillageCatDad** [comment]
The Organize button doesn't have a menu, it changes the color of the feed list (left pane) and allows you to move stuff around.  It's a toggle button.  The Options button has a menu.  When I click on Export... there, I get a "Save As" dialog box.  When I click on Import, I get a file selection dialog box.  Not sure what is happening on your system.

Putting the feeds into folders allows you to better organize your feeds, like into categories or topics.  You can click on the folder, and see only the unread items from the feeds in that folder.  For example, I have folders for "News: National", "News: Local", "News: Space", "Humor", etc.

Exporting to a file means your feed list is in a file that can be backed up.  That way, if Brief stops working due to a FF profile problem (for example), once you've fixed your FF profile, you can re-install Brief, and import your feed list in one ingest.  You will have lost your reading history, but at least you don't have to manually select one feed at a time all over again.  And if you use folders, your organization is restored as well.

**Comment by rob-stewart** [comment]
On February 5, 2019 6:42:23 PM EST, VillageCatDad <notifications@github.com> wrote:
> The Organize button doesn't have a menu, it changes the color of the
> feed list (left pane) and allows you to move stuff around.  It's a
> toggle button.  The Options button has a menu.  When I click on
> Export... there, I get a "Save As" dialog box.  When I click on
> Import, I get a file selection dialog box.  Not sure what is happening
> on your system.
> 
> Putting the feeds into folders allows you to better organize your
> feeds, like into categories or topics.  You can click on the folder,
> and see only the unread items from the feeds in that folder.  For
> example, I have folders for "News: National", "News: Local", "News:
> Space", "Humor", etc.
> 
> Exporting to a file means your feed list is in a file that can be
> backed up.  That way, if Brief stops working due to a FF profile
> problem (for example), once you've fixed your FF profile, you can
> re-install Brief, and import your feed list in one ingest.  You will
> have lost your reading history, but at least you don't have to
> manually select one feed at a time all over again.  And if you use
> folders, your organization is restored as well.  
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/brief-rss/brief/issues/431#issuecomment-460849054

Did I say the Organize button? I did mean Options. Anyway, I don't get a Save As dialog, so I'm no better off than before.
--
Rob

(Sent from my portable computation device.)

**Comment by rob-stewart** [comment]
@VillageCatDad 

When I turn on the Organize mode and click on the ellipsis at the bottom, nothing happens. Brief is clearly not happy.

**Comment by VillageCatDad** [comment]
@rob-stewart 

When I turn on Organize mode and click on the ellipsis, I can immediately start typing the name of the new folder.  When I hit return, it's created, and a new ellipsis appears on the bottom.

**Comment by rob-stewart** [comment]
@VillageCatDad 

I didn't realize that I should just start typing where the ellipses appear. When I clicked on an ellipsis, there was no UI change, so I thought nothing happened. I am able to create folders in addition to those which I already have from when Brief used bookmarks to manage things.

Thanks for your help.

(BTW, nothing happens when I choose Export still. Sad.)

**Comment by harry-dresden** [comment]
My feeds won't open a window or export anything, either, though the import command opens a window. Any suggestions for making that work? I've already reinstalled Brief. Is there a limit to how many feeds can be exported? I have hundreds. Thanks.
